### DISCUSSION TITLE
Syntax highlighting in a code block with more than one annotation

### DISCUSSION THREAD

**Discussion by barafael** [marp-team/marp#443] [Q&A] [answered]
Say I have this code block:

<pre>
```rust, no_run, my-context

```
</pre>

In this case, syntax highlighting does not work. If `rust` is the only thing after <pre>```</pre>, then of course it does work.

Is this by design, or can it be configured somehow?

**Comment by yhatt** [comment, marked as answer]
[CommonMark](https://commonmark.org/), the normative spec of Markdown, treats [the first word that is following to the fence `` ``` `` as the language for syntax highlighting.](https://spec.commonmark.org/0.30/#example-142) `,` is not treated as a delimiter of the language so the code block uses the unknown `rust,` language to highlight, not `rust` language.

If possible, you should remove the comma following to `rust` language string, like this:

````
```rust no_run, my-context

```
````

See also: https://github.com/markdown-it/markdown-it/issues/626#issuecomment-877635706. This thread makes clear that `,` is not allowed in CommonMark.

**Reply by barafael** [reply]
Thank you for clarifying!
